package-res/reportviewer/reportviewer.js

```diff
@@ -70,6 +71,9 @@ define([ 'common-ui/util/util', 'common-ui/util/timeutil', 'common-ui/util/forma
         this.reportPrompt.api.event.afterUpdate(this.afterUpdateCallback.bind(this));
         this.reportPrompt.api.event.afterRender(this.view.afterRender.bind(this.view));
 
+        if (typeof parent.pho !== 'undefined' && typeof parent.pho.dashboards !== 'undefined') {
+          parent.pho.dashboards.addEditContentToggledListener(this.editModeToggledHandler.bind(this));
```

Comment:
one last thing: Every add should at some point have a remove, right? Otherwise, when the dashboard decides to remove your iframe, you will be left with a memory leak and/or an invalid function.

Have a look at how the mantle-handler does this job here, and do the same.